File: JavaScript_Courses/JavaScript_Basics/Preparation/05_While_Loop_Exercise/05_Coins.js
function coins(input) {
    let moneyAmount = Number(input[0]);
    let coinsCount = 0;

    let currentAmountInCents = Math.round(moneyAmount * 100);
    // let currentAmountInCents = moneyAmount * 100;
    // console.log(currentAmountInCents);

    while (currentAmountInCents > 0) {
        if (currentAmountInCents >= 200) {
            currentAmountInCents -= 200;
        } else if (currentAmountInCents >= 100) {
            currentAmountInCents -= 100;
        } else if (currentAmountInCents >= 50) {
            currentAmountInCents -= 50;
        } else if (currentAmountInCents >= 20) {
            currentAmountInCents -= 20;
        } else if (currentAmountInCents >= 10) {
            currentAmountInCents -= 10;
        } else if (currentAmountInCents >= 5) {
            currentAmountInCents -= 5;
        } else if (currentAmountInCents >= 2) {
            currentAmountInCents -= 2;
        } else if (currentAmountInCents >= 1) {
            currentAmountInCents -= 1;
        }
        coinsCount++;
    }
    console.log(coinsCount);
}


// coins(["1.23"]);
// coins(["2.49"]);
// coins(["1.51"]);

File: Python_Courses/Python_Basics/Python_PB_2024_10/05_While_Loop_Exercise/05_Coins.py
change = float(input())  # float('2.73') == 2.73

coins_count = 0
# while change > 0:
while True:
    if change <= 0:
        break

    if change >= 2:
        change -= 2
    elif change >= 1:
        change -= 1
    elif change >= 0.50:
        change -= 0.50
    elif change >= 0.20:
        change -= 0.20
    elif change >= 0.10:
        change -= 0.10
    elif change >= 0.05:
        change -= 0.05
    elif change >= 0.02:
        change -= 0.02
    elif change >= 0.01:
        change -= 0.01

    coins_count += 1
    change = round(change, 2)

print(coins_count)
